leftwm/src/config/mod.rs

```diff
@@ -123,6 +123,53 @@ impl Keybind {
     }
 }
 
+/// Selecting by `WM_CLASS` and/or window title, allow the user to define if a
+/// window should spawn on a specified tag and/or its floating state.
+///
+/// # Example
+///
+/// In `config.toml`
+///
+/// ```toml
+/// [[window_config_by_class]]
+/// wm_class = "krita"
+/// spawn_on_tag = 3
+/// spawn_floating = false
+/// ```
+///
+/// windows whose WM_CLASS is "krita" will spawn on tag 3 (1-indexed) and not floating.
+#[derive(Serialize, Deserialize, Default, Debug)]
+pub struct WindowHook {
+    /// `WM_CLASS` in X11
+    pub window_class: Option<String>,
```

Comment:
Hmm, would/should this be an enum between window class and name when we move away from TOML?

Reply:
It would be nicer on the code side, but how would that work in the config side? One advantage of having them separate is it allows you to be more specific, e.g. you much a terminal by class but only the ones with a certain name.

Reply:
It would work the same with the current scoring; the `WindowHook` matching the title would be selected over the one with the class.

Having an Enum

```rust
struct WindowMatch {
    Class(String),
    Title(String),
}

struct WindowHook {
    wm_match: Option<WindowMatch>,
    // ...
}
```

would be fine. A custom implementation for `Deserialize` would be needed as of now, though.